src/Attributes.php

```diff
@@ -518,4 +511,66 @@ public function getIterator(): Traversable
     {
         return new ArrayIterator($this->attributes);
     }
+
+    /**
+     * Rebind all callbacks that point to `$oldThisId` to `$newThis`
+     *
+     * @param int    $oldThisId
+     * @param object $newThis
+     */
+    public function rebind(int $oldThisId, object $newThis): void
+    {
+        $this->rebindCallbacks($this->callbacks, $oldThisId, $newThis);
+        $this->rebindCallbacks($this->setterCallbacks, $oldThisId, $newThis);
+    }
+
+    /**
+     * Loops over all `$callbacks`, binds them to `$newThis` only where `$oldThisId` matches. The callbacks are
+     * modified directly on the `$callbacks` reference.
+     *
+     * @param callable[] $callbacks
+     * @param int        $oldThisId
+     * @param object     $newThis
+     */
+    private function rebindCallbacks(array &$callbacks, int $oldThisId, object $newThis): void
+    {
+        foreach ($callbacks as &$callback) {
+            if (! $callback instanceof Closure) {
+                if (is_array($callback) && ! is_string($callback[0])) {
+                    if (spl_object_id($callback[0]) === $oldThisId) {
+                        $newClosure = Closure::fromCallable($callback);
+                        $methodName = (new ReflectionFunction($newClosure))->name;
+
+                        $reflectedClosure = new ReflectionMethod($callback[0], $methodName);
+                        if ($reflectedClosure->isStatic()) {
+                            continue;
+                        }
+
+                        $callback = $newClosure->bindTo($newThis);
```

Comment:
```suggestion
                        $callback[0] = $newThis;
```

Or has this a disadvantage I'm not aware of?